src/components/PlacementResult.js

```diff
@@ -34,6 +35,22 @@ export default class PlacementResult extends React.Component<PlacementResultDefa
 
   static displayName = 'PlacementResult';
 
+  static contextTypes = {
+    searcher: PropTypes.any,
+  };
+
+  onPromotionClick() {
+    const { linkUrl, linkText, imageUrl } = this.props;
+    const searcher = this.context.searcher;
+    const addPromotionSignal = searcher ? searcher.addPromotionSignal : null;
+    if (!addPromotionSignal) {
+      return;
+    }
+    const signalDocId = imageUrl ? `${imageUrl}-${linkUrl || ''}` : `${linkText || ''}-${linkUrl || ''}`;
```

Comment:
I'd add a comment here saying explicitly how you're building the "doc ID" from the promotion's properties. Anyone wanting to make use of these signals will need to know how to parse it to get those pieces back...